packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -137,22 +135,50 @@ function s(
     if (
         !isUndefined(slotset) &&
         !isUndefined(slotset[slotName]) &&
-        slotset[slotName].length !== 0
+        slotset[slotName].vnodes.length !== 0
     ) {
-        children = slotset[slotName];
+        aChildren = slotset[slotName];
     }
-    const vmBeingRendered = getVMBeingRendered()!;
-    const { renderMode, shadowMode } = vmBeingRendered;
+
+    const { renderMode, shadowMode } = getVMBeingRendered()!;
 
     if (renderMode === RenderMode.Light) {
-        sc(children);
-        return children;
+        // In LightDom, the effective children are returned as is, we will need to
+        // diff the children (patchChildren) using an empty that is always present (to get the parentElement) and
+        // the same in both collection to patch (so it serves as an anchor).
+
+        // Note on why 2: if the same vnode is reused, it may lead to issues when removing because of elm.(1 vnode for 2 elements)
```

Comment:
@nolanlawson @pmdartus, I hit this issue while trying to reuse the same vnode for the markers, in order to fix #2688, we will also need to think about what to do about `vnode.elm`.

Reply:
Do we need to create the markers here? 
Could those markers be created in the `mountSlot` when needed?

Reply:
What benefits do you see from creating them in `mountSlot` vs. here? In this logic, the markers will be added only when needed.

I also think that if we add them in `mountSlot`, then we need to repeat the logic in the hydration.